src/solc_0.8/asset/AssetSignedAuctionAuthV2.sol

```diff
@@ -0,0 +1,520 @@
+        address signer;
+
+        if (eip712) {
+            dataToHash = abi.encodePacked(
+                "\x19\x01",
+                _DOMAIN_SEPARATOR,
+                _hashAuction(from, token, auctionData, ids, amounts)
+            );
+        } else {
+            dataToHash = _encodeBasicSignatureHash(from, token, auctionData, ids, amounts);
+        }
+
+        if (signatureType == SignatureType.EIP1271) {
+            require(
+                ERC1271(from).isValidSignature(dataToHash, signature) == ERC1271_MAGICVALUE,
+                "invalid 1271 signature"
+            );
+        } else if (signatureType == SignatureType.EIP1654) {
+            require(
+                ERC1654(from).isValidSignature(keccak256(dataToHash), signature) == ERC1654_MAGICVALUE,
+                "invalid 1654 signature"
+            );
+        } else {
+            signer = SigUtil.recover(keccak256(dataToHash), signature);
+            require(signer == from, "signer != from");
+        }
+
+        return signer;
+    }
+
+    function _encodeBasicSignatureHash(
+        address from,
+        address token,
+        uint256[] memory auctionData,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) internal view returns (bytes memory) {
+        return
+            SigUtil.prefixed(
+                keccak256(
+                    abi.encodePacked(
+                        address(this),
+                        AUCTION_TYPEHASH,
+                        from,
+                        token,
+                        auctionData[AuctionData_OfferId],
+                        auctionData[AuctionData_StartingPrice],
+                        auctionData[AuctionData_EndingPrice],
+                        auctionData[AuctionData_StartedAt],
+                        auctionData[AuctionData_Duration],
+                        auctionData[AuctionData_Packs],
+                        keccak256(abi.encodePacked(ids)),
+                        keccak256(abi.encodePacked(amounts))
+                    )
+                )
+            );
+    }
+
+    function _hashAuction(
+        address from,
+        address token,
+        uint256[] memory auctionData,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) internal pure returns (bytes32) {
+        return
+            keccak256(
+                abi.encode(
+                    AUCTION_TYPEHASH,
+                    from,
+                    token,
+                    auctionData[AuctionData_OfferId],
+                    auctionData[AuctionData_StartingPrice],
+                    auctionData[AuctionData_EndingPrice],
+                    auctionData[AuctionData_StartedAt],
+                    auctionData[AuctionData_Duration],
+                    auctionData[AuctionData_Packs],
+                    keccak256(abi.encodePacked(ids)),
+                    keccak256(abi.encodePacked(amounts))
+                )
+            );
+    }
+}
```

Comment:
need to add tests for the auth

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/617/commits/64329bbf20f10643091a4e37167754b5d187d48d